Automattic/sensei
# Add FSE support for course theme

Fixes #4512, fixes #4582, fixes #4586
Partially #4595

### Changes proposed in this Pull Request

- Add hooks and filters to make course theme templates editable in the Site Editor
  - Overrides theme for the site editor page and adds it to the admin menu if the site is not currently using a block theme
  - Adds just the two templates if the site is already running a block theme
- Move the lesson and quiz templates to HTML files. 
  - Add compatibility functions to the theme to load the HTML templates on WordPress versions where block templates are not available (<5.9)
- Add content styles and global styles configuration from Blockbase's ponyfill.css base
- Add basic JS components for the course theme blocks

### Testing instructions

- Enable the feature flag: `add_filter( 'sensei_feature_flag_course_theme, '__return_true' );`
- Update to Wordpress 5.9-RC1. Disable Gutenberg plugin. Activate a non-block theme.
- Navigate to Appearance > Editor in WP-Admin
- Select the `Lesson (Learning Mode)` template to edit
- Move things around in the template and save
- Open a lesson using course theme and check that the edits were applied
- Make sure everything else works as before
---
- Activate a block theme
- Make sure the Lesson and Quiz templates show up in the site editor, and edits are applied
---
- Switch to a site with Wordpress 5.7 or 5.8
- Make sure Learning Mode is still working on the frontend. (Not editable for these versions)


### Screenshot / Video

https://user-images.githubusercontent.com/176949/149145835-3fc7ad90-d076-45e5-aa3e-11af553a6c2f.mov

Showing up as plugin templates when using a block theme: 

<img width="1705" alt="image" src="https://user-images.githubusercontent.com/176949/149145950-d8064ba3-cc5e-44d3-bbc1-495b4a751b49.png">
